src/s390_ecc.c

```diff
@@ -598,12 +602,96 @@ ECDSA_SIGN_REPLY* make_ecdsa_sign_request(const ICA_EC_KEY *privkey,
 	offset = make_cprbx((struct CPRBX *)cbrbmem, parmblock_len, preqcblk, prepcblk);
 	offset += make_ecdsa_sign_parmblock((ECDSA_PARMBLOCK_PART1*)(cbrbmem+offset), hash, hash_length);
 	offset += make_keyblock_length((ECC_KEYBLOCK_LENGTH*)(cbrbmem+offset), keyblock_len);
-	offset += make_ecdsa_private_key_token(cbrbmem+offset, privkey, curve_type);
+	offset += make_ecdsa_private_key_token(cbrbmem+offset, privkey, X, Y, curve_type);
 	finalize_xcrb(xcrb, preqcblk, prepcblk);
 
     return (ECDSA_SIGN_REPLY*)prepcblk;
 }
 
+/**
+ * calculate the public (X,Y) values for the given private key, if necessary.
+ */
+unsigned int provide_pubkey(const ICA_EC_KEY *privkey, unsigned char *X, unsigned char *Y)
+{
+	EC_KEY *eckey = NULL;
+	EC_POINT *pub_key = NULL;
+	const EC_GROUP *group = NULL;
+	BIGNUM *bn_d = NULL, *bn_x = NULL, *bn_y = NULL;
+	char* x_str = NULL;
+	char* y_str = NULL;
+	int privlen = -1;
+	unsigned int i, n, rc;
+
```

Comment:
I can't see the point why here all the variables get an initialization value. I checked the BIGNUM, privlen and eckey variable and their first use is as lvalue so no init needed.

Reply:
Agreed. I will remove the initializations (and also go over the complete code again to find similar places).